Library/Homebrew/rubocops/checksum_cop.rb

```diff
@@ -0,0 +1,54 @@
+require_relative "./extend/formula_cop"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      class Checksum < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          return if body_node.nil?
+          if method_called_ever?(body_node, :md5)
+            problem "MD5 checksums are deprecated, please use SHA256"
+          end
+
+          if method_called_ever?(body_node, :sha1)
+            problem "SHA1 checksums are deprecated, please use SHA256"
+          end
+
+          sha256_calls = find_every_method_call_by_name(body_node, :sha256)
+          sha256_calls.each do |sha256_call|
+            sha256_node = get_checksum_node(sha256_call)
+            audit_sha256(sha256_node)
+          end
+        end
+
+        def get_checksum_node(call)
+          return if parameters(call).empty? || parameters(call).nil?
+          if parameters(call).first.str_type?
+            parameters(call).first
+          elsif parameters(call).first.hash_type?
```

Comment:
Might want to document with a comment that this is the case for `bottle do` blocks.